modules/ingest-common/src/main/java/org/opensearch/ingest/common/FingerprintProcessor.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.ingest.common;
+
+import org.opensearch.common.Nullable;
+import org.opensearch.common.hash.MessageDigests;
+import org.opensearch.ingest.AbstractProcessor;
+import org.opensearch.ingest.ConfigurationUtils;
+import org.opensearch.ingest.IngestDocument;
+import org.opensearch.ingest.Processor;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static org.opensearch.ingest.ConfigurationUtils.newConfigurationException;
+
+/**
+ * Processor that generating hash value for the specified fields or all fields in a document
+ */
+public class FingerprintProcessor extends AbstractProcessor {
+    public static final String TYPE = "fingerprint";
+    private static final Set<String> HASH_METHODS = Set.of("MD5", "SHA-1", "SHA-256");
```

Comment:
I believe MD5 and SHA-1 are considered weak and insecure, any reasons to have them? SHA-256 is fine, probably worth adding and SHA-3 (https://openjdk.org/jeps/287).

Reply:
> For now, we only have 3 algorithms in the common module, do you think we should add more?

I think at least SHA3 would make sense, since it has direct support in JDK. Wdyt?

Reply:
Sure, I've added SHA3-256, please help to take a look, thank you!